### ISSUE TITLE
PHP7.2 email piping error

### ISSUE THREAD

**Issue by matuscak** [osTicket/osTicket#4777]
### Prerequisites

[*] Can you reproduce the problem in a fresh installation of the "develop" branch?
Yes, it is the same when using latest commit from develop-next 

[*] Do you have any errors in the PHP error log, or javascript console?
No, but there are two error messages written to osTicket SYSLOG on each invocation:

API Unexpected Data
system_emails: Unexpected data received in API request

API Unexpected Data
thread_entry_recipients: Unexpected data received in API request

[*] Did you check the [osTicket forums](http://osticket.com/forum/categories)?

[*] Did you [perform a cursory search](https://github.com/osTicket/osTicket/issues) to see if your bug or enhancement is already reported?

### Description
With PHP 5.6 email piping works as expected. 
When I switch PHP to version 7.2 (and everything else remains the same) delivering emails to api/pipe.php start failing, pipe.php ends with exit code 255 (reported by postfix) and two error messages mentioned above are written to osTicket syslog table.

I know that PHP 7.2 was not yet supported month or two ago but after release of 1.11 it should be supported now, or not?

### Versions

osTicket Version | v1.11-23-gd5f5825d (d5f5825)     —  Up to date
Web Server Software | Apache/2.4.38 (FreeBSD) PHP/7.2.15
MySQL Version | 5.6.43
PHP Version | 7.2.15

gdlib | Used for image manipulation and PDF printing
imap | Used for email fetching
xml | XML API
xml-dom | Used for HTML email processing
json | Improves performance creating and processing JSON
mbstring | Highly recommended for non western european language content
phar | Highly recommended for plugins and language packs
intl | Highly recommended for non western european language content
fileinfo | Used to detect file types for uploads
APCu | Improves overall performance
Zend Opcache | Improves overall performance

cgi.fix_pathinfo | "1" is recommended if AJAX is not working
date.timezone | Europe/Berlin

Schema | ost (localhost)
Schema Signature | 00c949a623b82848baaf3480b51307e3
Space Used | 4.53 MiB
Space for Attachments | 0.08 MiB
Timezone | PST                      (Interpreted as America/Los_Angeles)

**Comment by JediKev** [comment]
@matuscak 

When and where did you download osTicket from? You should be using the download from our [official site](https://osticket.com/download) or latest pull from the `develop-next` branch here. You will encounter some issues via pipe/fetching/etc. but all email issues are addressed with these:
- https://github.com/osTicket/osTicket/pull/4758
- https://github.com/osTicket/osTicket/pull/4731

**P.S.**
If you use the download from our site then you must fix the file permissions recursively before you attempt to deploy them to production.

Cheers.

**Comment by matuscak** [comment]
I'm tracking develop-next branch published on Github and doing updates with 
./manage.php deploy -vC /www/ost-root-folder

I did a fresh "git pull" now and there was one last commit I was missing so currently reported version is:
v1.11-25-ge60b5c0a (e60b5c0)

Behaviour is the same. With PHP 5.6 email piping works, with PHP 7.2 it fails.
If you need some additional debug data, tell me what shoulf I capture. Thanks.

**Comment by JediKev** [comment]
@matuscak 

Some sample data would be nice.

Cheers.

**Comment by matuscak** [comment]
Sample data of what? Email being sent?

It fails on any e-mail, content does not matter. And if I just switch to PHP 5.6 (on the same server) and resend the same e-mail, it works.

This is what is written in postfix mail log:
Mar  6 13:05:54 <mail.info> ost postfix/pipe[44845]: 1085A2731F: to=<ost11@ost11.local>, orig_to=<ost11@ost.int.next-works.com>, relay=ost11, delay=1.9, delays=0.09/0.01/0/1.8, dsn=5.3.0, status=bounced (Command died with status 255: "/usr/local/bin/php")

Command being run by postfix is:
/usr/local/bin/php /www/ost/api/pipe.php

But the same postfix with same config works with PHP 5.6 so I believe it cannot be postfix configuration issue.

**Comment by JediKev** [comment]
@matuscak 

Nah, I don't think it's postfix either. Did you deploy the two pull requests to your install and retest?

Cheers.

**Comment by matuscak** [comment]
Sorry, I thought they are already merged in latest develop-next, when you were asking where did I download it.

So now I tried to apply both and #4731 seems to solve the issue. Thanks.
